---
title: "Tajimas D Analysis"
author: "Murray Cadzow"
date: "12/03/2015"
output: html_document
---



```{r, fig.height=9, fig.width=12, warning=FALSE}
library(dplyr)
library(ggplot2)
library(scales)
library(readr)

POP=""
TD=data.frame()
s=data.frame()
make_graph=function(TD,s){
  ggplot(data = TD, aes(x=BIN_START, y=TajimaD), ylim=c(min(TD$TajimaD -0.5), max(TD$TajimaD + 0.5)))  + 
    geom_point(shape = 16, colour= alpha("black", 1/5)) + 
    #geom_hex() +
    facet_wrap(~CHROM, scales = "free_x")  + 
    geom_hline(aes(yintercept= q1, colour ="quantile"), data=s,)  + 
    geom_hline(aes(yintercept= q2, colour ="quantile"), data=s)  +
    geom_hline(aes(yintercept= mean, colour="mean"), data=s) + 
    scale_colour_manual("",breaks=c("mean","quantile"),values=c("green","red")) + 
    scale_x_continuous( xlab("Chromosome Position (Mbp)")) + 
    ylab("Tajima's D") + 
    ggtitle(paste0(POP," Tajima's D by Chromosome")) + 
    theme( plot.background= element_rect(colour="black",fill=NA), legend.position= c(0.75, 0.12)) + 
    theme_bw()
}
setwd("~/MurrayXsan/Bioinformatics/working_dir/extract/TajD/")

get_genes_in_region= function(regions, size){
  library(RMySQL)
  drv = dbDriver("MySQL")
  ensembl_core = dbConnect(drv, user="anonymous", host="ensembldb.ensembl.org", dbname="homo_sapiens_core_75_37", password="")
  for(i in 1:length(regions[,1])){
    chr= as.data.frame(regions)[i,1]
    pos1= as.data.frame(regions)[i,2]
    pos2 = pos1+size
    
    print(dbGetQuery(ensembl_core, paste0("select s.name, g.seq_region_start, g.seq_region_end, x.display_label, s.coord_system_id from  gene g, seq_region s, xref x where s.name =", chr, " AND (", pos1 ," > g.seq_region_start AND ",pos2,"  < g.seq_region_end OR ",pos1," BETWEEN g.seq_region_start AND g.seq_region_end OR ",pos2," BETWEEN g.seq_region_start AND g.seq_region_end)  AND g.display_xref_id = x.xref_id group by x.display_label order by s.name *1, g.seq_region_start")))
  }
}

for( POP in c("AXIOM","OMNI","CEU","CHB","CHS","GBR","YRI")){
  print(POP)
  
  TD=data.frame()
  for( i in 1:22){
    TD=rbind(TD,read.table(file = paste0(POP,i,".taj_d"), header=TRUE))
  }
  s = TD %>% group_by(CHROM) %>% summarise(mean=mean(TajimaD), sd=sd(TajimaD), min=min(TajimaD), max=max(TajimaD), q1 = quantile(TajimaD, 0.01), q2 = quantile(TajimaD, 0.99))
  print(as.data.frame(round(s, digits=2)))
  plot(make_graph(TD,s))
  top = TD %>% arrange(TajimaD) %>% head(n=10)
    #filter(min(TajimaD) == TajimaD | max(TajimaD) == TajimaD) 
  print(as.data.frame(top))
  get_genes_in_region(as.data.frame(top), size=30000)
}

```
